rpcd/playbooks/roles/elasticsearch/tasks/upgrade_elasticsearch_pre.yml

```diff
@@ -0,0 +1,201 @@
+    port: 9200
+    state: started
+    timeout: 300
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Drop log_test1 template
+  shell: 'curl -sk -XDELETE {{ internal_lb_vip_address }}:{{ elasticsearch_http_port }}/_template/log_test1'
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+    - skip_ansible_lint
+
+- name: Place legacy liberty template script
+  template:
+    src: legacy-mapping.sh.j2
+    dest: /opt/legacy-mapping.sh
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install legacy liberty template
+  shell: '/opt/legacy-mapping.sh'
+  register: legacy_template_result
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Remove legacy liberty template script
+  file:
+    name: '/opt/legacy-mapping.sh'
+    state: absent
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install Reindexing Plugin
+  command: ./plugin -install {{ item }}
+  args:
+    chdir: /usr/share/elasticsearch/bin
+    creates: /usr/share/elasticsearch/plugins/reindexing
+  with_items:
+    - org.codelibs/elasticsearch-reindexing/1.7.0
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install reindexing wrapper script
+  copy:
+    src: reindex-liberty.py
+    dest: /opt/reindex-liberty.py
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+    - reindex-wrapper
+
+- name: Reindex legacy elasticsearch indices
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --reindex
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Monitor reindexing process
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --monitor --continuous
+  async: 259200
+  poll: 5
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Drop legacy elasticsearch indeices
+  shell: /opt/reindex-liberty.py --host "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}" --drop
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-reindex
+
+- name: Remove Reindexing Plugin
+  command: ./plugin -remove {{ item }}
```

Comment:
``./plugin remove {{ item }}``, without the dash.
Otherwise it failed for me:
```
root@cantu-testup-logging2-elasticsearch-container-749fc2d4:/usr/share/elasticsearch/bin# ./plugin -remove org.codelibs/elasticsearch-reindexing/1.7.0
ERROR: unknown command [-remove]. Use [-h] option to list available commands
```